cmd/fs-v1.go

```diff
@@ -346,6 +390,15 @@ func (fs fsObjects) CopyObject(srcBucket, srcObject, dstBucket, dstObject string
 		// Return the new object info.
 		return fsMeta.ToObjectInfo(srcBucket, srcObject, fi), nil
 	}
+	if srcEtag != "" {
+		etag, perr := fs.getObjectETag(srcBucket, srcObject)
+		if perr != nil {
+			return oi, toObjectErr(perr, srcBucket, srcObject)
+		}
+		if etag != srcEtag {
+			return oi, toObjectErr(errors.Trace(InvalidETag{}), srcBucket, srcObject)
+		}
+	}
```

Comment:
since fs.getObject() is getting called below which also checks for srcEtag, you need not do it here

Reply:
actually this check is needed, but it should be done before `if cpMetadataOnly {...}`
so you can pass `""` to getObject() so that it does not verify it again

Reply:
done.